sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -756,35 +756,42 @@ ProgramManager::collectDependentDeviceImagesForVirtualFunctions(
     }
   }
 
-  while (!WorkList.empty()) {
-    std::string SetName = WorkList.front();
-    WorkList.pop();
+  {
```

Comment:
We may be able to save the mutex lock if there was nothing in the work list.
```suggestion
  if (!WorkList.empty()) {
```